src/abstractdataframe/subset.jl

```diff
@@ -0,0 +1,257 @@
+# subset allows a transformation specification without a target column name or a column
+
+_process_subset_pair(i::Int, a::ColumnIndex) = a => Symbol(:x, i)
+_process_subset_pair(i::Int, @nospecialize(a::Pair{<:Any, <:Base.Callable})) =
+    first(a) => last(a) => Symbol(:x, i)
+_process_subset_pair(i::Int, a) =
+    throw(ArgumentError("condition specifier $a is not supported by `subset`"))
+
+@noinline function _get_subset_conditions(df::Union{AbstractDataFrame, GroupedDataFrame},
+                                @nospecialize(args), skipmissing::Bool)
+    conditions = Any[_process_subset_pair(i, a) for (i, a) in enumerate(args)]
+
+    isempty(conditions) && throw(ArgumentError("at least one condition must be passed"))
+
+    if df isa AbstractDataFrame
+        df_conditions = select(df, conditions..., copycols=!(df isa DataFrame))
+    else
+        df_conditions = select(df, conditions...,
+                               copycols=!(parent(df) isa DataFrame), keepkeys=false)
+    end
+    for col in eachcol(df_conditions)
+        if !(eltype(col) <: Union{Missing, Bool})
+            throw(ArgumentError("each transformation must produce a vector whose " *
+                                "eltype is subtype of Union{Missing, Bool}"))
+        end
+    end
+
+    @assert ncol(df_conditions) == length(conditions)
+
+    if ncol(df_conditions) == 1
+        cond = df_conditions[!, 1]
+    else
+        cond = .&(eachcol(df_conditions)...)
+    end
+
+    @assert eltype(cond) <: Union{Bool, Missing}
+
+    if skipmissing
+        return coalesce.(cond, false)
+    else
+        # actually currently the inner condition is only evaluated if actually
+        # we have some missings in cond, but it might change in the future so
+        # I leave this check
+        if Missing <: eltype(cond)
+            pos = findfirst(ismissing, cond)
+            if pos !== nothing
+                # note that the user might have passed a GroupedDataFrame but we
+                # then referer to the parent data frame of this GroupedDataFrame
+                throw(ArgumentError("skipmissing=false and in row $pos of the" *
+                                    " passed conditions were evaluated to" *
+                                    " missing value"))
+            end
+        end
+        return cond
+    end
+end
+
+"""
+    subset(df::AbstractDataFrame, args...; skipmissing::Bool=false, view::Bool=false)
+    subset(gdf::GroupedDataFrame, args...; skipmissing::Bool=false, view::Bool=false)
+
+Return a copy of data frame `df` or parent of `gdf` containing only rows for
+which all values produced by transformation(s) `args` for a given row are `true`.
+
+If `skipmissing=true`, producing `missing` in conjunction of results produced by
+`args` is allowed and corresponding rows are dropped. If `skipmissing=false`
+(the default) an error is thrown if `missing` is produced.
```

Comment:
Maybe worth being even more explicit:
```suggestion
If `skipmissing=false` (the default), an error is thrown if the conjunction (i.e. `&`)
of results produced by `args` is `missing`. If `skipmissing=true`, `missing`
is allowed and corresponding rows are dropped. Note that if some transformations
in `args` return `false` and others `missing` for a given row, the conjunction is `false`,
but that if some return `true` and others `missing`, the conjunction is `missing`
([three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic)).
```